operator/pkg/ciliumidentity/cache_test.go

```diff
@@ -323,3 +329,72 @@ func TestCIDUsageInCES(t *testing.T) {
 	assert.Equal(t, 0, state.CIDUsageCount("1000"), assertTxt)
 	assert.Equal(t, 0, state.CIDUsageCount("2000"), assertTxt)
 }
+
+func TestEnqueueTimeTracker(t *testing.T) {
+	testCases := []struct {
+		name         string
+		item         string
+		setBeforeGet bool
+	}{
+		{"Set and get", "item1", true},
+		{"Get without set", "item2", false},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tracker := &EnqueueTimeTracker{enqueuedAt: make(map[string]time.Time)}
+
+			if tc.setBeforeGet {
+				tracker.SetEnqueueTimeIfNotSet(tc.item)
+			}
+
+			_, exists := tracker.GetEnqueueTimeAndReset(tc.item)
+			if exists != tc.setBeforeGet {
+				t.Errorf("Expected exists to be %v, got %v", tc.setBeforeGet, exists)
+			}
+
+			_, exists = tracker.GetEnqueueTimeAndReset(tc.item)
+			if exists {
+				t.Errorf("Expected exists to be false, got %v", exists)
+			}
+		})
+	}
+}
+
+func TestCIDDeletionTracker(t *testing.T) {
+	logger := slog.New(logging.SlogNopHandler)
+
+	testCases := []struct {
+		name             string
+		cidName          string
+		expectMarkedTime bool
+		unmarkBeforeGet  bool
+	}{
+		{"Mark and retrieve", "cid1", true, false},
+		{"Mark empty CID", "", false, false},
+		{"Unmark existing", "cid2", false, true},
+		{"Unmark non-existing", "cid3", false, true},
+	}
```

Comment:
Just as the previous test, it might be better to use two different sets of subtests:

- the first one for the subtests with `unmarkBeforeGet == false`
- the second one for the remaining subtests with `unmarkBeforeGet == true`

This should help in removing some ifs and flags and improving the readability

Reply:
refactored the test, splitted by method.

Reply:
refactored the test, splitted by method.